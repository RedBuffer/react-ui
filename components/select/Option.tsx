import * as React from 'react';
import classnames from 'classnames';

interface PropTypes
  extends Omit<
    React.DetailedHTMLProps<
      React.OptionHTMLAttributes<HTMLOptionElement>,
      HTMLOptionElement
    >,
    'size'
  > {
  children?: React.ReactNode;
}

const Component = ({
  children,
  className,
  ...restProps
}: PropTypes): React.ReactElement => {
  return (
    <option {...restProps} className={classnames(className)}>
      {children}
    </option>
  );
};

export default Component;
