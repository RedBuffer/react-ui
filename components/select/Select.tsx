import * as React from 'react';
import classnames from 'classnames';

interface PropTypes
  extends Omit<
    React.DetailedHTMLProps<
      React.SelectHTMLAttributes<HTMLSelectElement>,
      HTMLSelectElement
    >,
    'size'
  > {
  children?: React.ReactNode;

  size?: 'sm' | 'md' | 'lg';
}

const Component = ({
  children,
  className,
  size = 'md',
  ...restProps
}: PropTypes): React.ReactElement => {
  let padding: number;

  if (size === 'sm') {
    padding = 1;
  } else if (size === 'lg') {
    padding = 3;
  } else {
    padding = 2;
  }

  return (
    <select
      {...restProps}
      className={classnames(
        'px-0',
        'border',
        'w-full',
        'rounded',
        'shadow-sm',
        'transition',
        'ease-in-out',
        'duration-150',
        'text-default',
        `text-${size}`,
        `py-${padding}`,
        'leading-tight',
        'focus:outline-none',
        'focus:shadow-outline',
        'hover:border-blue-500',
        className,
      )}
    >
      {children}
    </select>
  );
};

export default Component;
