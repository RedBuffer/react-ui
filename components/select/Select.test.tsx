import * as React from 'react';
import { shallow, mount } from 'enzyme';

import Select from './Select';
import Option from './Option';

describe('Select', () => {
  let wrapper;
  let handleChange;

  beforeEach(() => {
    handleChange = jest.fn((event) => event.target.value);
    wrapper = mount(
      <Select onChange={handleChange}>
        <Option value={'option 1'}>Option 1</Option>
        <Option id={'option-2'} value={'option 2'}>
          Option 2
        </Option>
        <Option id={'option-3'} value={'option 3'}>
          Option 3
        </Option>
        <Option id={'option-4'} value={'option 4'}>
          Option 4
        </Option>
      </Select>,
    );
  });

  it('should render correctly', () => {
    const component = shallow(<Select />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with options', () => {
    const component = wrapper;

    expect(component).toMatchSnapshot();
  });

  it('select option 1 in select', () => {
    wrapper
      .find('select')
      .simulate('change', { target: { value: 'option 2' } });

    expect(handleChange.mock.calls.length).toBe(1);
  });
});
