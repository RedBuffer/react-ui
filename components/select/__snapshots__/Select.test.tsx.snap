// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `
<select
  className="px-0 border w-full rounded shadow-sm transition ease-in-out duration-150 text-default text-md py-2 leading-tight focus:outline-none focus:shadow-outline hover:border-blue-500"
/>
`;

exports[`Select should render correctly with options 1`] = `
<Component
  onChange={[MockFunction]}
>
  <select
    className="px-0 border w-full rounded shadow-sm transition ease-in-out duration-150 text-default text-md py-2 leading-tight focus:outline-none focus:shadow-outline hover:border-blue-500"
    onChange={[MockFunction]}
  >
    <Component
      value="option 1"
    >
      <option
        className=""
        value="option 1"
      >
        Option 1
      </option>
    </Component>
    <Component
      id="option-2"
      value="option 2"
    >
      <option
        className=""
        id="option-2"
        value="option 2"
      >
        Option 2
      </option>
    </Component>
    <Component
      id="option-3"
      value="option 3"
    >
      <option
        className=""
        id="option-3"
        value="option 3"
      >
        Option 3
      </option>
    </Component>
    <Component
      id="option-4"
      value="option 4"
    >
      <option
        className=""
        id="option-4"
        value="option 4"
      >
        Option 4
      </option>
    </Component>
  </select>
</Component>
`;
